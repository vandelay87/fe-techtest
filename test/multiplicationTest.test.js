import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import NumberList from '../src/numberList';

describe('numberList.js', () => {
  it('renders component', () => {
    const tree = shallow(<NumberList />);

    expect(tree).toMatchSnapshot();
  });

  it("renders array of 10 items", () => {
    const wrapper = shallow(<NumberList limit={30} />);

    expect(wrapper.find('div').children().length).toBe(30);
  });

  it("highlights multiplication numbers when clicked", () => {
    const wrapper = shallow(<NumberList limit={4} />);

    wrapper.find('div').children().at(2).simulate("click");
    // expect style to be highlighted style

    wrapper.find('div').children().at(2).simulate("click");
    // expect style to NOT be highlighted style
  });
});
