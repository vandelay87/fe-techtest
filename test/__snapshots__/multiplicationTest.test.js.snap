// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiplicationTest.js renders component 1`] = `ShallowWrapper {}`;
